tests/integration/standard/test_metadata.py

```diff
@@ -124,7 +124,10 @@ def test_bad_contact_point(self):
 
         @test_category metadata
         """
-        self.assertEqual(len(self.cluster.metadata.all_hosts()), 3)
+        for i in range(10):
+            for host in self.cluster.metadata.all_hosts():
+                self.assertNotEquals(host.endpoint.address, '126.0.0.186')
```

Comment:
maybe add a busy loop waiting until `self.cluster.metadata` contains 3 elements and then compare it's content with expected value? IMO it would be much cleaner than this approach. I can't tell what this tests validates without reading the description.

Reply:
that test just add a list of hosts, and expecting to filter the non-existing one (the last one):
```
self.cluster = TestCluster(contact_points=['127.0.0.1', '127.0.0.2', '127.0.0.3', '126.0.0.186'])
```

Reply:
I'm aware of that, just can't deduce it without reading the description in test setup. Also this tests takes always 5s, compared to <=5s with suggested approach.

Reply:
FYI setup of 3 node cluster, takes much more ~ 4min in this case. so 5s is small :)

Anyhow I've found the actually root cause, and is that I've missed updating the hosts for peers, so in cases it host_id didn't match they are thrown away, while they shouldn't.

Reply:
> FYI setup of 3 node cluster, takes much more ~ 4min in this case. so 5s is small :)

Add constant 5s 50times and you tests will always take 4min each. Minimizing test lenght is a good practice you cannot battle with. 

What's the point of review if pull requests get merged without any approval nor resolving review remarks?

Reply:
the end result test is doing what you suggested

https://github.com/scylladb/python-driver/commit/e44785df89c4d63fc12898e645b9f1734f34ae33 is the core change that fixes the test.